code/datums/components/shell.dm

```diff
@@ -252,6 +254,29 @@
 	remove_circuit()
 	return ITEM_INTERACT_BLOCKING
 
+/**
+ * Called when a wrench is used on the parent, if it has the SHELL_FLAG_REQUIRE_ANCHOR flag.
+ * Toggles parent anchoring state if not locked.
+ */
+/datum/component/shell/proc/on_wrench_act(atom/source, mob/user, obj/item/tool)
+	SIGNAL_HANDLER
+	if(!is_authorized(user))
+		return
+
+	if(!isstructure(parent))
+		return
+
+	var/obj/structure/struct = parent
+	if(locked)
+		if(shell_flags & SHELL_FLAG_ALLOW_FAILURE_ACTION)
+			return
+		source.balloon_alert(user, "it's locked!")
+		return ITEM_INTERACT_BLOCKING
+	struct.set_anchored(!struct.anchored)
+	tool.play_tool_sound(struct)
+	source.balloon_alert(user, struct.anchored ? "secured" : "unsecured")
+	return ITEM_INTERACT_BLOCKING
```

Comment:
make sure that doors cannot be unanchored.
The reason the logic was done this way was to let objects control when they get anchored but I understand why you're doing it this way.

Just make sure that the object can be anchored.

Reply:
_Nod_. I think I tested this, and noted it's not an issue because they're not structures code-wise, but let me test it again right now.
I do feel that, yes, it should be possible to let the objects override the anchoring logic, but _right now_ I'm not sure how to do that without it being awkward like the previous solution. Let me think about it.

Reply:
Oh, I think I see the issue, right.
Circuit airlocks work fine, because they're not structures but _machinery_:
![image](https://github.com/tgstation/tgstation/assets/42909981/bfde5446-46b6-49f1-bc2f-eb818da2e20a)
This matches my notes on previous tests.

HOWEVER, _the door assembly_ is a structure, right? The structure you get when manually building an airlock.
But even though the assembly is a structure, the resulting door is machinery, and it doesn't actually apply the shell component to the assembly, only to the resulting door. So we get:
![image](https://github.com/tgstation/tgstation/assets/42909981/2c9bc9d0-dc6e-475d-856a-016101956ac2)
So that seems to work fine too.